### ISSUE TITLE
Getting Redis maxRetriesPerRequest error since 10.3.1

### ISSUE THREAD

**Issue by u12206050** [directus/directus#19038]
### Describe the Bug

Since deploying 10.3.1, and updating REDIS env variables as described. We have been getting 503 errors during requests to the api.

Logs in the backend show it to be a Redis issue with error `MaxRetriesPerRequestError: Reached the max retries per request limit (which is 20)`.

We are using GCP with Memory store Redis. Worked fine before 10.3.1.

### To Reproduce

Upgrade and deploy 10.3.1

Make some api requests and look out for 503 erros

### Directus Version

10.3.1

### Hosting Strategy

Self-Hosted (Custom)

**Comment by rijkvanzanten** [comment]
Did you use the same Redis connection for all 4 different configuration points before? Beyond the env var names, nothing else of the setup changed yet 🤔

**Comment by u12206050** [comment]
Yes, so migrated to using REDIS only.

I just redeployed now, adding back `CACHE_REDIS` and now it seems to work again. Very strange, since I can't even find `CACHE_REDIS` in the directus code on latest main. Could release have been without it?

**Comment by BlackDahlia313** [comment]
> Yes, so migrated to using REDIS only.
> 
> I just redeployed now, adding back `CACHE_REDIS` and now it seems to work again. Very strange, since I can't even find `CACHE_REDIS` in the directus code on latest main. Could release have been without it?

Only used Cloud Run. But is there any build cache maybe hanging on that you could clear?

**Comment by rijkvanzanten** [comment]
That particular error most often means that it wasn't able to connect to Redis at all, which to me indicates that either the correct env vars weren't available in the latest version, or it was secretly still running the old containers while using the new env vars 🤔

**Comment by u12206050** [comment]
Thanks for the tip, although after investigating I can confirm that the errors got triggered on the new containers with the updated code. I removed the CACHE_REDIS again now (after 20 hours since the last deployment), and know for sure I am running the latest code, and yet again the `MaxRetriesPerRequestError` error reappears.

So I started debugging it myself, and since I couldn't find any trace of `CACHE_REDIS` in the directus main branch, I search through my node_modules directory that gets installed when I install `directus`

Turns out there is two `@directus/api` being installed. One is being installed in the `node_modules` folder my project, and the other is being installed as a dependency of `directus`. But since my only dependency is `directus` I had to find where the other version comes from.

Turns out that some custom extension had `@directus/api` as a peer dependency on version `^10.2.0` which at first seems like that should be fine, however then I saw that the actual "latest" version was `12.0.0`, so I am assuming that even though it is just a peer dependency that for some reason the 10.2.0 version still gets installed, and in the end is actually used somehow.
